datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -86,7 +86,11 @@ pub enum AggregateMode {
 #[derive(Clone, Debug, Default)]
 pub struct PhysicalGroupBy {
     /// Distinct (Physical Expr, Alias) in the grouping set
-    expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
+    grouping_set_expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
+    /// Hidden grouping set expr in the grouping set
+    hidden_grouping_set_expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
+    /// Distinct result expr for the grouping set, used to generate output schema
+    result_expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
```

Comment:
I have a question.

Is `result_expr` a part of `grouping_set_expr`?
If yes, maybe we can use a index vec to point `grouping_set_expr`